server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java

```diff
@@ -95,38 +110,147 @@ public void optionDownload(String username) throws Throwable {
             with.header("Authorization", accessToken.serialize());
         }
 
-        response = with.options("/download/myBlob");
+        response = with
+                .options("/download/" + ONE_ATTACHMENT_EML_ATTACHEMENT_BLOB_ID);
     }
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\"$")
     public void downloads(String username, String attachmentId) throws Throwable {
         String blobId = blobIdByAttachmentId.get(attachmentId);
+        response = requestWithAuthentication(blobId, username)
+                .get("/download/" + blobId);
+    }
+
+    private RequestSpecification requestWithAuthentication(String blobId, String username) {
         AccessToken accessToken = userStepdefs.tokenByUser.get(username);
-        RequestSpecification with = with();
+        AttachmentAccessTokenKey key = new AttachmentAccessTokenKey(username, blobId);
+        if (attachmentAccessTokens.containsKey(key)) {
+            return with().param("access_token", attachmentAccessTokens.get(key).serialize());
+        }
         if (accessToken != null) {
-            with.header("Authorization", accessToken.serialize());
+            return with().header("Authorization", accessToken.serialize());
+        }
+        return with();
+    }
+
+    @When("^\"([^\"]*)\" is trusted for attachment \"([^\"]*)\"$")
+    public void attachmentAccessTokenFor(String username, String attachmentId) throws Throwable {
+        userStepdefs.connectUser(username);
```

Comment:
In order to be trusted, we have to connect first, then access to the POST method.
Then the sentence:
`is trusted for attachment`
is more logic when testing the GET method than:
`is connected`
`is trusted for attachment`

As the user doesn't have to be `connected`.